File: lib/common/models/android/pose_estimation_settings.dart
class AndroidPoseEstimationSettings {
  int? poseCount;
  int? animationDuration;
  String? faceNotInside;
  String? faceNotStraight;
  String? faceIsTooFar;
  String? holdPhoneVertically;
  String? alertTitle;
  String? alertDescription;
  String? alertTryAgain;
  String? turnLeft;
  String? turnRight;
  String? turnUp;
  String? turnDown;
  String? lookStraight;
  bool? mainGuideVisibility;
  bool? secondaryGuideVisibility;

  AndroidPoseEstimationSettings(
      {this.poseCount,
      this.animationDuration,
      this.faceNotInside,
      this.faceNotStraight,
      this.faceIsTooFar,
      this.holdPhoneVertically,
      this.alertTitle,
      this.alertDescription,
      this.alertTryAgain,
      this.turnLeft,
      this.turnRight,
      this.turnUp,
      this.turnDown,
      this.lookStraight,
      this.mainGuideVisibility,
      this.secondaryGuideVisibility});

  Map toJson() {
    return {
      "poseCount": poseCount,
      "animationDuration": animationDuration,
      "faceNotInside": faceNotInside,
      "faceNotStraight": faceNotStraight,
      "faceIsTooFar": faceIsTooFar,
      "holdPhoneVertically": holdPhoneVertically,
      "alertTitle": alertTitle,
      "alertDescription": alertDescription,
      "alertTryAgain": alertTryAgain,
      "turnLeft": turnLeft,
      "turnRight": turnRight,
      "turnUp": turnUp,
      "turnDown": turnDown,
      "lookStraight": lookStraight,
      "mainGuideVisibility": mainGuideVisibility,
      "secondaryGuideVisibility": secondaryGuideVisibility,
    };
  }
}

File: android/src/main/kotlin/ai/amani/flutter_amanisdk/modules/config_models/PoseEstimationSettings.kt
package ai.amani.flutter_amanisdk.modules.config_models
import  ai.amani.flutter_amanisdk.modules.JSONConvertable
import kotlinx.parcelize.Parcelize

@Parcelize
class PoseEstimationSettings(
        val poseCount: Int,
        val animationDuration: Int,
        val faceNotInside: String,
        val faceNotStraight: String,
        val faceIsTooFar: String,
        val holdPhoneVertically: String,
        val alertTitle: String,
        val alertDescription: String,
        val alertTryAgain: String,
        val turnLeft: String,
        val turnRight: String,
        val turnUp: String,
        val turnDown: String,
        val lookStraight: String,
        val mainGuideVisibility: Boolean = true,
        val secondaryGuideVisibility: Boolean = true,
): JSONConvertable